### ISSUE TITLE
Regression: Train::PlatformDetectionFailed: Sorry, we are unable to detect your platform

### ISSUE THREAD

**Issue by vsingh-msys** [inspec/train#538]
# Version:
3.1.10

# Environment:
Chef Infra Client: 15.5.17

# Scenario:
Trying to bootstrap a ubuntu node which configured with `requiretty`. Seems the problem arises after https://github.com/inspec/train/commit/1eb18d490a0cf0a60cfb5754ee44d36bffabbf10 commit.

# Steps to Reproduce:

1. Spin a Ubuntu vagrant node and set requiretty by adding `Defaults requiretty` in the sudoers file.
2. Bootstrap the node

```
knife bootstrap node1.test -N node1.test -i .vagrant/machines/default/virtualbox/private_key -U vagrant --sudo --yes -VV
```

# Expected Result:
It should bootstrap the node

# Actual Result:
```
Train::PlatformDetectionFailed: Sorry, we are unable to detect your platform
 /train/lib/train/platforms/detect/scanner.rb:41:in `scan'
 /train/lib/train/platforms/detect.rb:9:in `scan'
 /train/lib/train/plugins/base_connection.rb:116:in `platform'
 /train/lib/train/extras/command_wrapper.rb:166:in `load'
 /train/lib/train/transports/ssh_connection.rb:53:in `initialize'
 /train/lib/train/transports/ssh.rb:240:in `new'
 /train/lib/train/transports/ssh.rb:240:in `create_new_connection'
 /train/lib/train/transports/ssh.rb:82:in `connection'
 /chef/lib/chef/knife/bootstrap/train_connector.rb:57:in `connection'
 /chef/lib/chef/knife/bootstrap/train_connector.rb:70:in `connect!'
 /chef/lib/chef/knife/bootstrap.rb:692:in `do_connect'
 /chef/lib/chef/knife/bootstrap.rb:620:in `connect!'
 /chef/lib/chef/knife/bootstrap.rb:569:in `run'
 /chef/lib/chef/knife.rb:486:in `block in run_with_pretty_exceptions'
 /chef/lib/chef/local_mode.rb:42:in `with_server_connectivity'
 /chef/lib/chef/knife.rb:485:in `run_with_pretty_exceptions'
 /chef/lib/chef/knife.rb:229:in `run'
 /chef/lib/chef/application/knife.rb:163:in `run'
 /chef/bin/knife:24:in `<top (required)>'
 /chef/vendor/bundle/ruby/2.6.0/bin/knife:23:in `load'
 /chef/vendor/bundle/ruby/2.6.0/bin/knife:23:in `<top (required)>'

```

**Comment by miah** [comment]
Is there a specific version of Ubuntu that this is failing against, or all Ubuntu?

**Comment by vsingh-msys** [comment]
It will fail for all system which is configured for `pty` request. I have tested on ubuntu 16.04.

**Comment by christian-wtd** [comment]
What concerns me more than my code not working is the fact that it works without.
I was just enforcing LinuxCommand Wrapper on sudo/shell option to make sudo also effective for detection commands.
Basically "requiretty" should also prevent any other command executions with sudo on that host via train as long as it does not create a virtual tty which from my point of view breaks the meaning of "requiretty" on sudo as this should prevent anyone from running sudo remotely with scripts but only interactively.

**So there are two questions left:**
1) What does train to work around the "requiretty" restriction and
2) Why this work around is missing when enforcing LinuxCommand Wrapper

**Comment by zenspider** [comment]
This has been addressed via #544.
